src/client/Microsoft.Identity.Client/TokenSource.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Specifies the source of the access and Id tokens in the authentication result.
+    /// </summary>
+    public enum TokenSource
+    {
+        /// <summary>
+        /// The source of access and Id token is not set.
+        /// </summary>
+        None = 0,
+        /// <summary>
+        /// The source of access and Id token is MSAL's cache.
+        /// </summary>
+        Cache,
+        /// <summary>
+        /// The source of the access and Id token is Identity Provider - Azure Active Directory (AAD), ADFS or AAD B2C.
+        /// </summary>
+        IdentityProvider,
+        /// <summary>
+        /// The source of the access and Id token is a broker application - Authenticator or Company Portal. Brokers are supported only on Android and iOS.
+        /// </summary>
+        Broker
```

Comment:
There are no unit tests that cover this. While the brokers are really only implemented in mobile code, which is not testable, it would be valuable to have a test that proves that the TokenSource value is indeed copied from MsalTokenSource to AuthenticationResult. I believe there are (a rather few) tests that mock IBroker, such as https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/ad7970c3904d25662e5d756d076f777b0ed1867a/tests/Microsoft.Identity.Test.Unit.net45/RequestsTests/BrokerRequestTests.cs#L183 - you can use this approach to unit test this.

Reply:
Actually this test is a better fit: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/029e6031a1bccf88c5e31154712fcaa60a1fb4dd/tests/Microsoft.Identity.Test.Unit.net45/RequestsTests/InteractiveRequestOrchestrationTests.cs#L106